File: root/log/log.js
// npm install mathjs
var math = require('mathjs');
// npm install sprintf
var sprintf = require('sprintf').sprintf;

var vals = [1, 2.7182818284, 10, 100, 300];

vals.forEach(function(v) {
    var s = sprintf('%.5f: %.5f -- %.5f', v,
        math.log(v, math.e), math.log(v, 10.0) );
    console.log(s);
});

File: root/log/log.c
/*
$ gcc -o lg log.c -lm
*/

#include <stdio.h>
#include <math.h>

void test()
{
    double vals[] = {1, 2.7182818284, 10, 100, 300};
    int i;

    for (i=0; i<5; i++)
    {
        printf("%.5f: %.5f -- %.5f\n", vals[i],
            log(vals[i]), log10(vals[i]));
    }
}

int main()
{
    test();
    return 0;
}
